### DISCUSSION TITLE
Prese e pannello energia

### DISCUSSION THREAD

**Discussion by magomer** [indomus/forum#706] [HUB: Home Assistant]
Ciao a tutti. 
Ho una presa con monitoraggio consumi integrata con local tuya che genera questi 4 sensor (con la scelta dei valori 1 - 18 -19 - 20 nelle impostazioni di local tuya) : 
switch - current - power - voltage
Riesco a fare il monitoraggio instantaneo con varie card di lovelance ma non capisco cosa fare per visualizzare questi sensori nel pannello energia o, ad esempio, con grafana .....
Qualcuno riesca a darmi indicazioni passo passo ?
Grazie.

**Comment by Scialla** [comment]
puoi vedere solo le entità con attributo device_class: energy e devono avere state_class: measurement o state_class: total_increasing. Inoltre devono avere come unit_of_measurement: kWh

se mancano questi attributi non li vedi nel pannello energy

**Comment by magomer** [comment]
Ciao.
Grazie per la risposta.
Ho appena montato uno shelly 2.5 e, in effetti, il pannello Energia me lo
vede in automatico. Quindi non ho proprio nessuna possibilità di vedere
quella presa, non so con la creazione di sensor o modificando qualche
attributo ?
Grazie.
Ciao.

Il giorno ven 10 set 2021 alle ore 22:59 Scialla ***@***.***>
ha scritto:

**Reply by Scialla** [reply]
certo che puoi, guarda sui dev tools quali attributi ha la tua entità e, basandoti su quello che ho scritto prima, customizza l'entità aggiungendo gli attributi che mancano (in customize.yaml)

**Reply by magomer** [reply]
Approfitto ancora di te.
L'entità che mi da il controllo istantaneo del consumo si chiama : sensor.home:power
Nel file customize ho scritto 
sensor.home_power:
  unit_of_measurement: kWh
  device_class: energy
  state_class: measurement
  friendly_name: HOME_energy

ma niente da fare ....

gli altri sensor dell'entità sono
sensor.home_current
senror.home_voltage

Dove sbaglio ?
Grazie ancora !

p.s.: logicamente nel config ho aggiunto  

homeassistant:
  customize: !include customize.yaml

**Reply by Scialla** [reply]
non puoi usare power, quella è la misura istantanea dell'assorbimento; tu devi usare una entità che vada a crescere, ovvero una entità energia che cresca nel tempo e non che dia un valore istantaneo; per quello servono i kWh

Se non hai il sensore energy puoi usare integration per creartelo dal power
https://www.home-assistant.io/integrations/integration/

**Reply by magomer** [reply]
Intanto grazie per la pazienza.
Leggo e smanetto un  pò....
Grazie !

**Reply by magomer** [reply]
Niente da fare, probabilmente non fa per me....
Ti spiego cosa ho fatto e poi abbandono....
File configuration come scritto nel link che mi hai girano

sensor:
  - platform: integration
    source: sensor.home_power   questo è il mio sensore power in W
    name: energy_spent
    unit_prefix: k
    round: 2

customize

sensor.energy_spent:
  state_class: measurement
  device_class: energy
  unit_of_measurement: kWh

Pannello energia non lo vede....

Grazie comunque.
Ciao

**Reply by Scialla** [reply]
ma l'entità sensor.energy_spent la vedi?
su dev tools ha gli attributi corretti?
ah devi aggiungerla nel recorder!!!
